packages/plugin-connector/src/data-sources/get-sources.js

```diff
@@ -100,6 +101,26 @@ export const getSources = async (sourcesDir) => {
 	return datasourceSpecs;
 };
 
+const hash_location = './.evidence/template/.evidence-queries/sources/hashes.json';
+
+/**
+ * Gets the hashes of all source files, at the time of their last execution.
+ * @returns {Promise<Record<string, Record<string, string | null>>>}
+ */
+export async function getPastSourceHashes() {
+	const hashes = await fs.readFile(hash_location, 'utf-8').catch(() => '{}');
+	return JSON.parse(hashes);
```

Comment:
This is likely worth writing a zod schema for

Reply:
How come? We're the ones controlling what's written to the file and TS would complain if we mess it up anywhere